### PR TITLE
perf: avoid microtick when not locked

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,7 +35,9 @@ export default class Mutex {
 
 	async withLock(task) {
 		try {
-			await this.lock();
+			if (!this.tryLock()) {
+				await this.lock();
+			}
```
Comment (ronag) [inline index.js:40]:
```suggestion
			}

```

File: index.js [hunk 2]
```diff
@@ -4,6 +4,15 @@ export default class Mutex {
 	#queue = new Queue();
 	#isLocked = false;
 
+	tryLock() {
```
Comment (sindresorhus) [inline index.js:7]: This either needs to be properly documented or made private.

File: index.js [hunk 3]
```diff
@@ -26,7 +35,10 @@ export default class Mutex {
 
 	async withLock(task) {
 		try {
-			await this.lock();
+			if (!this.tryLock()) {
```
Comment (sindresorhus) [inline index.js:38]: Add a short code comment about why we do this, so it won't be removed in the future.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
`tryLock` needs a test.

**Comment by sindresorhus** [comment]
Why close?

**Pull request closed without merging**